### PR TITLE
RWA-3295: Added staging env check for receivedMessages api

### PR DESCRIPTION
**Pull request by adityadwadasi**
**Before creating a pull request make sure that:**

- [ ] commit messages are meaningful and follow good commit message guidelines
- [ ] README and other documentation has been updated / added (if needed)
- [ ] tests have been updated / new tests has been added (if needed)

Please remove this line and everything above and fill the following sections:

### JIRA link (if applicable) ###
https://tools.hmcts.net/jira/browse/RWA-3295

### Change description ###

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[ ] No
```

### CODE REVIEW

File: src/main/java/uk/gov/hmcts/reform/wacaseeventhandler/controllers/ReceivedMessagesHealthController.java [hunk 1]
```diff
@@ -114,6 +120,12 @@ private boolean isDateWithinWorkingHours(LocalDateTime localDateTime) {
     }
 
     private boolean isNotEnabledForEnvironment(String env) {
+        if (ENV_AAT.equals(env)) {
+            URI currentUri = ServletUriComponentsBuilder.fromCurrentRequestUri().build().toUri();
```
Comment (raghera) [inline src/main/java/uk/gov/hmcts/reform/wacaseeventhandler/controllers/ReceivedMessagesHealthController.java:124]: could this ever be `null` or empty?
If it can then would it throw an exception?

Comment (raghera) [reply src/main/java/uk/gov/hmcts/reform/wacaseeventhandler/controllers/ReceivedMessagesHealthController.java:124]: Also should put some logging so we can see it is getting the correct url and excluding staging correctly when running.

File: src/main/java/uk/gov/hmcts/reform/wacaseeventhandler/controllers/ReceivedMessagesHealthController.java [hunk 2]
```diff
@@ -114,6 +120,12 @@ private boolean isDateWithinWorkingHours(LocalDateTime localDateTime) {
     }
 
     private boolean isNotEnabledForEnvironment(String env) {
```
Comment (raghera) [inline src/main/java/uk/gov/hmcts/reform/wacaseeventhandler/controllers/ReceivedMessagesHealthController.java:122]: Obviously we'd need a unit test to check this in various different scenarios.
I'd suggest making this method `public` too to make it easier to test in isolation.  Later on we can refactor it to be in it's own class.

File: src/main/java/uk/gov/hmcts/reform/wacaseeventhandler/controllers/ReceivedMessagesHealthController.java [hunk 3]
```diff
@@ -113,7 +119,18 @@ private boolean isDateWithinWorkingHours(LocalDateTime localDateTime) {
...
+        if (ENV_AAT.equals(env)) {
+
+            URI currentUri = ServletUriComponentsBuilder.fromCurrentRequestUri().build().toUri();
+            String currentUriString = "";
+            if (null != currentUri) {
+                currentUriString = currentUri.toString();
+            }
+            if (currentUriString.contains(STAGING_TEXT)) {
+                return true;
+            }
```
Comment (raghera) [inline src/main/java/uk/gov/hmcts/reform/wacaseeventhandler/controllers/ReceivedMessagesHealthController.java:132]: I don't think you need the currentUriString var since you have currentUri.toString?

Comment (raghera) [reply src/main/java/uk/gov/hmcts/reform/wacaseeventhandler/controllers/ReceivedMessagesHealthController.java:132]: Please can you out put the url string so we can see what it is in logs as mentioned above please?

File: src/test/java/uk/gov/hmcts/reform/wacaseeventhandler/controllers/ReceivedMessagesHealthControllerTest.java [hunk 4]
```diff
@@ -191,6 +197,48 @@ void test_health_reports_success_if_messages_check_disabled_in_current_environme
...
+    }
+
+    @ParameterizedTest
+    @MethodSource(value = "disabledEnvProvider")
+    void test_enabled_for_environment_return_true_demo_ithc(String env) {
+
+        setField(receivedMessagesHealthController, "receivedMessageCheckEnvEnabled", "prod,aat");
+
+        MockHttpServletRequest mockHttpServletRequest = new MockHttpServletRequest();
+        mockHttpServletRequest.setServerName("case-event-handler.any.environment");
```
Comment (raghera) [inline src/test/java/uk/gov/hmcts/reform/wacaseeventhandler/controllers/ReceivedMessagesHealthControllerTest.java:222]: Could the server name every be `null` or empty string?  If it is always guaranteed to be present then that's ok as is but if not then you could add a test where it is empty too.

File: src/test/java/uk/gov/hmcts/reform/wacaseeventhandler/controllers/ReceivedMessagesHealthControllerTest.java [hunk 5]
```diff
@@ -191,6 +197,48 @@ void test_health_reports_success_if_messages_check_disabled_in_current_environme
...
+
+        MockHttpServletRequest mockHttpServletRequest = new MockHttpServletRequest();
+        mockHttpServletRequest.setServerName("case-event-handler.staging.aat");
+        mockHttpServletRequest.setRequestURI("/health");
+
+        RequestContextHolder.setRequestAttributes(new ServletRequestAttributes(mockHttpServletRequest));
+
+        assertTrue(receivedMessagesHealthController.isNotEnabledForEnvironment("aat"));
+    }
+
```
Comment (raghera) [inline src/test/java/uk/gov/hmcts/reform/wacaseeventhandler/controllers/ReceivedMessagesHealthControllerTest.java:241]: A general comment since we are using the Http Request Context from Spring we should have a Spring Integration test that loads the Spring context.  That way you will know that the host name is present etc within the context.  

Here we are obviously just mocking all this.

File: src/test/java/uk/gov/hmcts/reform/wacaseeventhandler/controllers/ReceivedMessagesHealthControllerTest.java (hunk 4 above)
Comment (adityadwadasi) [reply src/test/java/uk/gov/hmcts/reform/wacaseeventhandler/controllers/ReceivedMessagesHealthControllerTest.java:223]: This method is called only in HTTP context. I added an unit test to expect exception its called outside the context .

File: src/main/java/uk/gov/hmcts/reform/wacaseeventhandler/controllers/ReceivedMessagesHealthController.java (hunk 3 above)
Comment (adityadwadasi) [reply src/main/java/uk/gov/hmcts/reform/wacaseeventhandler/controllers/ReceivedMessagesHealthController.java]: I added the logs Ravi.

**Review by raghera — APPROVED** [review]

File: src/test/java/uk/gov/hmcts/reform/wacaseeventhandler/controllers/ReceivedMessagesHealthControllerTest.java [hunk 6]
```diff
@@ -191,6 +198,57 @@ void test_health_reports_success_if_messages_check_disabled_in_current_environme
...
+        mockHttpServletRequest.setRequestURI("/health");
+
+        RequestContextHolder.setRequestAttributes(new ServletRequestAttributes(mockHttpServletRequest));
+
+        assertFalse(receivedMessagesHealthController.isNotEnabledForEnvironment(env));
+    }
+
+    @ParameterizedTest
+    @MethodSource(value = "disabledEnvProvider")
+    void test_enabled_for_environment_return_true_demo_ithc(String env) {
```
Comment (BenNzewi) [inline src/test/java/uk/gov/hmcts/reform/wacaseeventhandler/controllers/ReceivedMessagesHealthControllerTest.java:218]: Test name contradicts test purpose. Should the name be test_not_enabled_for_environment_return_true_demo_ithc?

File: src/test/java/uk/gov/hmcts/reform/wacaseeventhandler/controllers/ReceivedMessagesHealthControllerTest.java [hunk 7]
```diff
@@ -191,6 +198,57 @@ void test_health_reports_success_if_messages_check_disabled_in_current_environme
...
+        mockHttpServletRequest.setServerName("case-event-handler.any.environment");
+        mockHttpServletRequest.setRequestURI("/health");
+
+        RequestContextHolder.setRequestAttributes(new ServletRequestAttributes(mockHttpServletRequest));
+
+        assertTrue(receivedMessagesHealthController.isNotEnabledForEnvironment(env));
+    }
+
+    @Test
+    void test_enabled_for_environment_return_true_staging_aat() {
```
Comment (BenNzewi) [inline src/test/java/uk/gov/hmcts/reform/wacaseeventhandler/controllers/ReceivedMessagesHealthControllerTest.java:232]: Test name contradicts test purpose. Should the name be test_not_enabled_for_environment_return_true_staging_aat?

**Review by raghera — APPROVED** [review]
